src/main/java/frc/robot/subsystems/signaling/Signaling.java

```diff
@@ -0,0 +1,229 @@
+  private PowerDistribution pdp;
+
+  public Signaling(Vision vision, PowerDistribution power) {
+    tick = 0;
+    patternTick = 0;
+    visionSubsystem = vision;
+    pdp = power;
+    candle.configBrightnessScalar(SignalingConstants.kLEDBrightness);
+  }
+
+  @Override
+  public void periodic() {
+    // Update Light Pattern
+
+    if (DriverStation.isDisabled()) {
+      tagCount = visionSubsystem.getTagCount();
+
+      switch ((int) tagCount) {
+        case 0:
+          setCandle(RGB.RED);
+          Logger.recordOutput("Signaling/CANdle Color", "Red");
+          break;
+        case 1:
+          setCandle(RGB.YELLOW);
+          Logger.recordOutput("Signaling/CANdle Color", "Yellow");
+          break;
+        default:
+          setCandle(RGB.GREEN);
+          Logger.recordOutput("Signaling/CANdle Color", "Green");
+          break;
+      }
+    } else {
+      if (getProblem()) {
+        setCandle(RGB.RED);
+      } else {
+        if (DriverStation.isAutonomous()) {
+          setCandle(RGB.PURPLE);
+
+        } else if (DriverStation.isEStopped()) {
+          setCandle(RGB.RED);
+        } else if (DriverStation.isTest()) {
+          setCandle(RGB.BLUE);
+        } else {
+          setCandle(RGB.GREEN);
+        }
+      }
+    }
+    Logger.recordOutput("Signaling/Pattern", disablePattern.toString());
+  }
+
+  private boolean getProblem() {
+    return (pdp.getVoltage() < 6) || (pdp.getTemperature() > 60);
+  }
+
+  public Command setColor(RGB rgb) {
+    return runOnce(
+        () -> {
+          setFullStrip(rgb);
+          Logger.recordOutput("Signaling/LED Color", rgb.toHex());
+        });
+  }
+
+  private void setCandle(RGB rgb) {
+    SmartDashboard.putString("Signaling/CANdle Color", rgb.toHex());
+    setSection(rgb, 0, 8);
+  }
+
+  public Command setToAlliance() {
+    return startEnd(
+        () -> {
+          setFullStrip(getColorFromAlliance(Constants.kAllianceColor));
+        },
+        this::resetLEDs);
+  }
+
+  public RGB getColorFromAlliance(Alliance alliance) {
+    if (alliance == Alliance.Red) {
+      return RGB.RED;
+    } else if (alliance == Alliance.Blue) {
+      return RGB.BLUE;
+    }
+    return RGB.BLUE;
+  }
+
+  private void setRumble(double intensity) {
+    OI.Driver.setRumble(intensity);
+    OI.Operator.setRumble(intensity);
+  }
+
+  public Command startSignal(RGB color) {
+    return startEnd(
+        () -> {
+          setFullStrip(color);
+          setRumble(1);
+        },
+        () -> {
+          resetLEDs();
+          setRumble(0);
+        });
+  }
+
+  private void resetLEDs() {
+    setFullStrip(RGB.BLACK);
+  }
+
+  private void setFullStrip(final RGB rgb) {
+    Logger.recordOutput("Signaling/LED Color", rgb.toHex());
+    setSection(rgb, 8, SignalingConstants.kNumLEDs);
+  }
+
+  private void setSection(final RGB rgb, final int startID, final int count) {
+    candle.setLEDs(rgb.red, rgb.green, rgb.blue, 0, startID, count);
+  }
```

Comment:
Try to add logic to avoid setting colors that the LEDs are already displaying. Especially with the long string of LEDs, since it may take a lot of our Command Loop time